.ci/releasenotes-gen-auto.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -e
+
+echo "PULL_REQUEST:""$PULL_REQUEST"
+if [[ $PULL_REQUEST =~ ^([0-9]+)$ ]]; then
+  echo "Build is not for Pull Request";
+  sleep 5;
+  exit 0;
+fi
+
+mkdir -p .ci-temp
+if [ -d .ci-temp/contribution ]; then
+  cd .ci-temp/contribution/
+  git reset --hard origin/master
+  git pull origin master
+  git fetch --tags
+  cd ../../
+else
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/contribution
+  cd ../
+fi
+cd .ci-temp/contribution/releasenotes-builder
+mvn -e --no-transfer-progress clean compile package
+cd ../../../
+
+# we need to do full clone as Travis do "git clone --depth=50"
+if [ -d .ci-temp/checkstyle ]; then
+  cd .ci-temp/checkstyle/
+  git reset --hard origin/master
+  git pull origin master
+  git fetch --tags
+  cd ../../
+else
+  cd .ci-temp/
+  git clone https://github.com/checkstyle/checkstyle
+  cd ../
+fi
+cd .ci-temp/checkstyle
+LATEST_RELEASE_TAG=$(git describe $(git rev-list --tags --max-count=1))
```

Comment:
Right now tags are only for releases.
But we plan to use them as trigger to do release presteps execution.
Let's find more reliable way to get latest released version.

Git is preferable way.
But if not possible to be done by pure git CLI, we can do request to GitHub API https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-10.3 to get latest release.

Reply:
We ned to update release note before release process is started. So we might need to trigger this script by tag - it will be latest tag. So latest tag will not be last released tag.

Reply:
Changes Done.

Command Used: `$(git describe --tags --abbrev=0)`

Reply:
>LATEST_RELEASE_TAG=$(git describe --tags --abbrev=0)

can you prove that this command will return latest  release tag ?
to test it: create tag "prepare-10.4.1" and execute your command.

Reply:
@romani can you please tell me how can I create a tag?

Reply:
No, Not this command but  `git describe --tags `git rev-list --tags --max-count=1`` can give the latest release.

Link to latest tag: https://github.com/Rahulkhinchi03/checkstyle/releases/tag/prepare-10.4.1

![avfgassdvds](https://user-images.githubusercontent.com/71710042/174506933-f9574da1-4027-4ebb-a3c7-8b598f09b029.png)

Reply:
@romani, this gist can help us to understand, what we really expect to get?
https://gist.github.com/rponte/fdc0724dd984088606b0

Reply:
Now using GitHub API

Reply:
I do not see changes, please push

Reply:
Please check now